### PR TITLE
Devtooling 286: Create New resource Organization Authentication Settings 

### PR DESCRIPTION
**Pull request by BrianMoyles**

### CODE REVIEW

File: genesyscloud/organization_authentication_settings/genesyscloud_organization_authentication_settings_init_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package organization_authentication_settings
```
Comment (dginty4) [inline genesyscloud/organization_authentication_settings/genesyscloud_organization_authentication_settings_init_test.go:1]: @carnellj Do we need an init_test file if there are no integration tests?

File: genesyscloud/organization_authentication_settings/resource_genesyscloud_organization_authentication_settings.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		log.Printf("Read organization authentication settings %s", d.Id())
+		return cc.CheckState()
+	})
+}
+
+// updateOrganizationAuthenticationSettings is used by the organization_authentication_settings resource to update an organization authentication settings in Genesys Cloud
+func updateOrganizationAuthenticationSettings(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := getOrgAuthSettingsProxy(sdkConfig)
+	AuthSettings := getOrganizationAuthenticationSettingsFromResourceData(d)
```
Comment (charliecon) [inline genesyscloud/organization_authentication_settings/resource_genesyscloud_organization_authentication_settings.go:70]: AuthSettings -> authSettings

File: genesyscloud/organization_authentication_settings/resource_genesyscloud_organization_authentication_settings_schema.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+			},
+			`domain_allowlist`: {
+				Description: `The list of domains that will be allowed to embed Genesys Cloud applications.`,
+				Optional:    true,
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			`ip_address_allowlist`: {
+				Description: `The list of IP addresses that will be allowed to authenticate with Genesys Cloud.`,
+				Optional:    true,
```
Comment (charliecon) [inline genesyscloud/organization_authentication_settings/resource_genesyscloud_organization_authentication_settings_schema.go:104]: Maybe we should put a warning in the description for `ip_address_allowlist` ? Describing the problem we had with this field

**Review by charliecon — APPROVED** [review]

File: genesyscloud/organization_authentication_settings/genesyscloud_organization_authentication_settings_init_test.go (hunk 1 above)
Comment (carnellj-genesys) [reply genesyscloud/organization_authentication_settings/genesyscloud_organization_authentication_settings_init_test.go:1]: No.  We can add one at a later date if we do not need one.

**Review by carnellj-genesys — CHANGES_REQUESTED** [review]

File: genesyscloud/organization_authentication_settings/resource_genesyscloud_organization_authentication_settings.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+)
+
+/*
+The resource_genesyscloud_organization_authentication_settings.go contains all of the methods that perform the core logic for a resource.
+*/
+
+func getAllOrganizationAuthenticationSettings(_ context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	resources := make(resourceExporter.ResourceIDMetaMap)
+	resources["0"] = &resourceExporter.ResourceMeta{Name: "organization_authentication_settings"}
```
[the remark this answers is not in the document]
Comment (BrianMoyles) [reply genesyscloud/organization_authentication_settings/resource_genesyscloud_organization_authentication_settings.go:26]: There is only 1 organization authentication settings so we can use the numerical value for the exporter

**Review by carnellj-genesys — APPROVED** [review]

**Review by dginty4 — APPROVED** [review]

**Pull request merged**